tests/Feature/Users/PasswordResetTest.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace Tests\Feature;
+
+use App\Notifications\ResetPassword;
+use App\User;
+use DB;
+use Illuminate\Support\Facades\Event;
+use Illuminate\Support\Facades\Notification;
+use Tests\TestCase;
+
+class PasswordResetTest extends TestCase
+{
+    public function testInvalidEmail() {
+        $response = $this->post('/user/recover', [
+            'email' => 'bademail!'
+        ]);
+
+        $response->assertSeeText(__('passwords.invalid'));
+    }
+
+    public function testUnknownEmail() {
+        $response = $this->post('/user/recover', [
+            'email' => 'nobody@nowhere.com'
+        ]);
+
+        $response->assertSeeText(__('passwords.user'));
+    }
+
+    public function testResetSuccess()
+    {
+        Notification::fake();
+        Event::fake();
+
+        $restarter = factory(User::class)->states('Restarter')->create();
+
+        $response = $this->post('/user/recover', [
+            'email' => $restarter->email
+        ]);
+
+        $response->assertSeeText(__('passwords.sent'));
+
+        Notification::assertSentTo(
+            [$restarter], ResetPassword::class
+        );
+
+        $restarter->refresh();
+
+        // Invalid code
+        $response = $this->post('/user/reset', [
```

Comment:
There is no assertion on $response before it is given a new value on line 57 - maybe these lines are redundant?